### ISSUE TITLE
podiff fails if trailing blank(s) outside of string delimiter

### ISSUE THREAD

**Issue by KennethNielsen** [pyg3t/pyg3t#12]
This impacts podiff, version 0.3.1.

If a trailing space is added to a line in either a msgid or msgstr outside of the string delimiter ("), then podiff fails with the following error message:

   ~/ % podiff a b
   Usage: podiff [OPTIONS] ORIGINAL_FILE UPDATED_FILE

   Use - as file argument to use standard in

   podiff: error: Cannot work with files with dissimilar base, unless the relax option (-r) or the full options (-f) is used.

   NOTE: This is not recommended..!
   Making a podiff for proofreading should happen between files with similar base, to make the podiff easier to read.

Steps to reproduce:

1) create two po-files (a and b) with identical content.
2 ) add a space to the end of a line in a msgid or msgstr to the right of that line's  final string delimiter (") in one of the files.
3) run podiff a b

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2012-01-16T17:03:43Z
 * owner: mbj
 * assignee: askhl
 * the launchpad url was https://bugs.launchpad.net/bugs/917259

**Comment by KennethNielsen** [comment]
(by k-nielsen81)
Thanks for reporting this. I can confirm the bug.

First of, podiff doesn't actually fail. The output you see is simply a warning, and it will still run if you give it the -r parameter (though it will give you a diff chunk for this message even though nothing has changed.)

I did a little checking and it turns out that the problem is actually in the parser. If I add a trailing space after e.g. this line:
msgid "Hello world!" 
in our testpofile.da.po file. Then the parser includes the last " in the msgid:
original file: ('Hello world!', None)
white space added: ('Hello world!"', None)
and this is the reason that podiff complains about dissimilar bases, because now it perceives it as if there has been a change in the msgids.

The second thing is to figure out whether this is actually a bug. It depends on whether or not such a po-file actually constitutes a  valid po-file (since we can't be expected to parse invalid files). Running both the original and the modified file through msgfmt -cv yeilds the same result and no warnings. We will have to consult the gettext documentation for further information, but since we have already encountered several file format questions that were not covered in the documentation (or where there is conflicts between the documentation and what msgfmt does), it is not certain that we will find the answer there and so we may have to fall back to saying that if msgfmt gives no warnings, that it is all right.

I'm assigning Ask to this bug, since he has been handling all the parser work.

Regards Kenneth
